File: leetcode/3248.py
class Solution:
    def finalPositionOfSnake(self, n: int, commands: List[str]) -> int:
        def f(pos,cmd):
            match cmd:
                case "UP":
                    return pos - n
                case "DOWN":
                    return pos + n
                case "RIGHT":
                    return pos + 1
                case "LEFT":
                    return pos - 1

        return reduce(f, commands, 0)

File: leetcode/3248.rs
fn main() {
    fn final_position_of_snake(n: i32, commands: Vec<String>) -> i32 {
        let f = |pos: i32, cmd: &String| -> i32 {
            pos + match &cmd[..] {
                "UP" => -n,
                "DOWN" => n,
                "RIGHT" => 1,
                "LEFT" => -1,
                _ => 0,
            }
        };

        commands.iter().fold(0, f)
    }

    println!("{}", final_position_of_snake(2, vec![String::from("RIGHT"), String::from("DOWN")])); // => 3
}
